doc/adding_to_an_adapter.md

```diff
@@ -1,120 +1,237 @@
+`type="7"`. To add the credential to the adapter instance, add an attribute `credentialKind` to the adapter instance's
+`ResourceKind` element, with a value of the `CredentialKind`'s `key` attribute.
 
-In the case where multiple credential kinds are supported, the keys are separated by a comma.
+### describe.xml
 ```xml
 <AdapterKind xmlns="http://schemas.vmware.com/vcops/schema" key="my_adapter" nameKey="1" version="1">
   <!--...-->
   <ResourceKinds>
-    <ResourceKind key="my_adapter_instance" credentialKind="my_credential_type,my_other_credential_type" nameKey="5" type="7" >
+    <ResourceKind key="my_adapter_instance" credentialKind="my_credential_type" nameKey="5" type="7" >
         <!-- ... -->
     </ResourceKind>
   </ResourceKinds>
 </AdapterKind>
 ```
 
-Once the credential is defined in the `describe.xml` file, it can be used in the adapter code.
-> Note: If there are any existing connections used by the [`mp-test`](mp-test.md) tool before the credential was created or updated, these will need to be deleted or updated.
-
-Once an adapter instance is defined, any credential fields (`CredentialField`) will be prompted to the user when creating an account in VMware Aria Operations on the `Data Sources` &rarr; `Integrations` page.
-After the account has been created, credential fields will be available in the input to the `collect`, `test_connection`, and `get_endpoint_urls` methods. (See [Creating an Adapter Instance](#creating-an-adapter-instance).)
-
-Using the [Python VMware Aria Operations Integration Module](python-integration-module.md), the credential is available in the input to the `collect`, `test_connection`, and `get_endpoint_urls` methods. (See [Creating an Adapter Instance](#creating-an-adapter-instance).)
+> ![Adding a credential](adding_a_credential.png)
+>
+> Adding a credential to an adapter instance with the `username` and `password` fields as defined above. 'Credential name'
+> is always added (by VMware Aria Operations), and allows for credentials to be reused between adapter instances.
+
+Both `describe.xml` and [AdapterDefinition](../lib/python/doc/aria/ops/definition/adapter_definition.html) allow for the
+use of multiple credential.
+
+> ### VMware Aria Operations Integration SDK library
+> ```python
+> def get_adapter_definition() -> AdapterDefinition:
+>     definition = AdapterDefinition("MyAdapter", "My Adapter")
+>
+>     credential = definition.define_credential_type("my_credential_type", "Credential")
+>     credential.define_string_parameter("user_name", "User Name")
+>     credential.define_password_parameter("user_password", "Password", required=False)
+>     # Second set of credentials
+>     second_credential = definition.define_credential_type("my_second_credential_type", "Second Credential")
+>     second_credential.define_string_parameter("second_user_name", "Second User Name")
+>     second_credential.define_password_parameter("second_user_password", "Second Password", required=False)
```

Comment:
I think if we do an example we should make sure it makes sense, so a user will have an example of _why_ they might need two credential kinds -- there's no point in having two credential kinds if they both have the same two pieces of information.